process/dataindexer/dataIndexer.go

```diff
@@ -30,16 +32,40 @@ func NewDataIndexer(arguments ArgDataIndexer) (*dataIndexer, error) {
 		return nil, err
 	}
 
+	blockContainer, err := createBlockCreatorsContainer()
```

Comment:
[comment]

Usually, we receive this as an input interface on the constructor.
Creation is usually outside

Reply:
moved to factory componnent